core/src/main/java/org/apache/carbondata/core/indexstore/BlockletDataMapIndexStore.java

```diff
@@ -234,6 +231,10 @@ private BlockletDataMapIndexWrapper get(TableBlockIndexUniqueIdentifierWrapper i
                 tableSegmentUniqueIdentifierWrapper.isAddTableBlockToUnsafeAndLRUCache());
       }
     }
+    String segmentPath =
+        tableSegmentUniqueIdentifierWrapper.getTableBlockIndexUniqueIdentifier().getIndexFilePath();
```

Comment:
`getIndexFilePath()` returns `segmentPath`?

Reply:
yes. getIndexFilePath() will return the indexFile parent path, which is segmentPath